kahuna/public/js/search-query/service.js

```diff
@@ -0,0 +1,32 @@
+import angular from 'angular';
+import Rx from 'rx';
+import '../util/rx';
+
+export const searchQueryService = angular.module('gr.searchQuery.service', ['util.rx']);
+
+searchQueryService.factory('searchQueryService', [function() {
+    const query$ = new Rx.Subject();
+
+    // TODO: When we update rx, we need to swap the seed / acc around
+    const q$ = query$.scan('', (state, fn) => fn(state).trim());
```

Comment:
Just to check: does `trim` not get in the way of typing several words (i.e. does it trim the trailing space while typing)?